workers/unity/Packages/com.improbable.gdk.core/Systems/CommandBookkeepingSystem.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Improbable.Gdk.Core.Commands;
+using Unity.Entities;
+
+namespace Improbable.Gdk.Core
+{
+    public class CommandBookkeepingSystem : ComponentSystem
+    {
+        private Dictionary<Type, CommandStorage> commandStorageInstances = new Dictionary<Type, CommandStorage>();
+
+        public TConcreteCommandStorage GetCommandStorageForType<TConcreteCommandStorage>()
+            where TConcreteCommandStorage : CommandStorage
+        {
+            if (!commandStorageInstances.TryGetValue(typeof(TConcreteCommandStorage), out var storage))
+            {
+                throw new ArgumentException(
+                    $"Could not find command storage of type {typeof(TConcreteCommandStorage).Name}");
+            }
+
+            return (TConcreteCommandStorage) storage;
+        }
+
+        protected override void OnCreateManager(int capacity)
+        {
+            base.OnCreateManager(capacity);
+            Enabled = false;
+            CreateConcreteCommandStorage();
+        }
+
+        // Will never tick - only need this to compile.
+        protected override void OnUpdate()
```

Comment:
Might want to disable on first tick then. Would also be good to see if there is another way to do this. Having a system that doesn't do anything is odd. I don't think there is another way that easily lets you inject something though. We could store stuff on a shared component on the worker entity maybe.

Reply:
Just realised we set enabled to false in OnCreateManager

Reply:
Its disabled in the `OnCreateManager`

Reply:
I think we can make it a `ScriptBehaviourManager` but there are a lot more overrides that you need